.github/workflows/pr-check.yml

```diff
@@ -0,0 +1,13 @@
+name: PR Check
```

Comment:
Could rename to build & test on also apply on main branches (with `on: push`)

Reply:
we don't want to build snapshots of PR branches though...

Reply:
this is why I added "on main branches". You can limit those, see eg https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags

Reply:
that's by workflow. how would you exclude the job/step of building a pr commit? isn't it far simpler to introduce a separate workflow on:push which targets maintenance branches/master, that will effectively test+build+publish snapshots?

Reply:
Ah, you're right, once we factor in the publication step, we're better off with a different workflow indeed. Sorry for the noise.